src/newmatrix.cpp

```diff
@@ -2398,24 +2427,24 @@ ACMD(do_connect)
   if (GET_OBJ_TYPE(jack) == ITEM_CYBERWARE) {
     if (GET_CYBERWARE_TYPE(jack) == CYB_DATAJACK) {
       if (GET_CYBERWARE_FLAGS(jack) == DATA_INDUCTION) {
-        snprintf(buf, sizeof(buf), "$n places $s hand over $s induction pad as $e connects to %s.",
+        snprintf(buf, sizeof(buf), "$n places $s hand over $s induction pad as $e connects to %s.\r\n",
           !proxy_deck && cyberdeck->obj_flags.extra_flags.IsSet(ITEM_EXTRA_OTAKU_RESONANCE) ? "the jackpoint" : "$s cyberdeck");
       } else {
-        snprintf(buf, sizeof(buf), "$n slides one end of the cable into $s datajack and the other into %s.",
+        snprintf(buf, sizeof(buf), "$n slides one end of the cable into $s datajack and the other into %s.\r\n",
           !proxy_deck && cyberdeck->obj_flags.extra_flags.IsSet(ITEM_EXTRA_OTAKU_RESONANCE) ? "the jackpoint" : "$s cyberdeck");
       }
     } else {
-      snprintf(buf, sizeof(buf), "$n's eye opens up as $e slides %s cable into $s eye datajack.",
+      snprintf(buf, sizeof(buf), "$n's eye opens up as $e slides %s cable into $s eye datajack.\r\n",
         !proxy_deck && cyberdeck->obj_flags.extra_flags.IsSet(ITEM_EXTRA_OTAKU_RESONANCE) ? "the jackpoint" : "$s cyberdeck");
     }
   } else {
-    snprintf(buf, sizeof(buf), "$n plugs the leads of $s 'trode net into $s cyberdeck.");
+    snprintf(buf, sizeof(buf), "$n plugs the leads of $s 'trode net into $s cyberdeck.\r\n");
   }
   act(buf, FALSE, ch, 0, 0, TO_ROOM);
   if (cyberdeck->obj_flags.extra_flags.IsSet(ITEM_EXTRA_OTAKU_RESONANCE)) 
-    act("You jack into the matrix to commune with the resonance.", FALSE, ch, 0, 0, TO_CHAR);
+    act("You jack into the matrix to commune with the resonance.\r\n", FALSE, ch, 0, 0, TO_CHAR);
```

Comment:
`act()` auto-appends `\r\n`. You only need to add it for `send_to_char()` calls.

edit: it looks like all of the messages you added \r\n to here are acts?

Reply:
I was encountering a problem despite that with room names appending on connection. I don't know if that's because of shared buffers or what, but the only fix was adding newlines?

Reply:
Nevermind I figured out why. Fixing